src/dropdown/dropdown.js

```diff
@@ -187,17 +187,30 @@ angular.module('ui.bootstrap.dropdown', ['ui.bootstrap.position'])
     }
   };
 
+  function toggleDropdownUp() {
```

Comment:
There should be no need to change the position service to accomplish this.  Calling the $position.positionElements with an auto placement should do what you need, just use the pos.top value returned.